src/auth/basic/Config.cc

```diff
@@ -75,14 +75,15 @@ Auth::Basic::Config::type() const
 void
```

Comment:
> PR description: Use the existing SchemeConfig::configured() method consistently to determine whether the scheme has been fully configured.

Please apply similar changes to _all_ contexts where authenticateProgram is used inconsistently. For example, Auth::SchemeConfig::dump() code assumes that authenticateProgram means "fully configured"[^1], but does not use configured(). Other examples include Auth::Basic::UserRequest::startHelperLookup() and Auth::Digest::UserRequest::addAuthenticationInfoHeader() AFAICT. I did not check all occurrences. Please fix all such cases in this PR.

[^1]: By using/reporting `authenticateChildren.n_max` and `realm` members without checking whether they were set/configured.